Pages with date as title (diary entries) cannot be processed. frontmatter does not match collection schema.

### Describe the bug

Hi, thanks for creating the plugin.

I have a standard obsidian layout with a `11_diary/` folder containing notes where the filename is the date on which they were created. 
When building it get the following error:
```
20:12:31 [ERROR] [vite] x Build failed in 8.18s
[InvalidContentEntryFrontmatterError] [astro:content-imports] docs → wiki/11_diary/2018-11-21.md frontmatter does not match collection schema.
title: Expected type `"string"`, received "date"
file: /home/caspar/dev/wiki_web/src/content/docs/wiki/11_diary/2018-11-21.md?astroContentCollectionEntry=true:0:0
  Hint:
    See https://docs.astro.build/en/guides/content-collections/ for more information on content schemas.
  Error reference:
    https://docs.astro.build/en/reference/errors/invalid-content-entry-frontmatter-error/
  Stack trace:
    at getEntryData (file:///home/caspar/dev/wiki_web/node_modules/astro/dist/content/utils.js:74:26)
    at async Object.transform (file:///home/caspar/dev/wiki_web/node_modules/astro/dist/content/vite-plugin-content-imports.js:89:67)
    at async ModuleLoader.addModuleSource (file:///home/caspar/dev/wiki_web/node_modules/rollup/dist/es/shared/node-entry.js:19003:36)
```

It looks like astro / vite parses the title as date instead of string. 

### To Reproduce

create a note with a filname that resembles a date (`2024-08-18.md`) and build it with starlight-obsidian. 

### Expected behavior

Page title should always be a string. 

### How often does this bug happen?

Every time

### System Info

_No response_

### Additional Context

_No response_

--- a/.github/workflows/integration.yml
+++ b/.github/workflows/integration.yml
@@ -18,12 +18,12 @@ jobs:
         uses: actions/checkout@v3
 
       - name: Install pnpm
-        uses: pnpm/action-setup@v2.2.4
+        uses: pnpm/action-setup@v4
         with:
           version: 8.14.1
 
       - name: Install Node.js
-        uses: actions/setup-node@v3
+        uses: actions/setup-node@v4
         with:
           cache: pnpm
           node-version: 18

--- a/.github/workflows/release.yml
+++ b/.github/workflows/release.yml
@@ -20,17 +20,17 @@ jobs:
       id-token: write
     steps:
       - name: Checkout
-        uses: actions/checkout@v3
+        uses: actions/checkout@v4
         with:
           fetch-depth: 0
 
       - name: Install pnpm
-        uses: pnpm/action-setup@v2.2.4
+        uses: pnpm/action-setup@v4
         with:
           version: 8.14.1
 
       - name: Install Node.js
-        uses: actions/setup-node@v3
+        uses: actions/setup-node@v4
         with:
           cache: pnpm
           node-version: 18

--- a/fixtures/basics/2024-08-19.md
+++ b/fixtures/basics/2024-08-19.md
@@ -1,0 +1,1 @@
+Test with a title looking like a date.

--- a/packages/starlight-obsidian/libs/remark.ts
+++ b/packages/starlight-obsidian/libs/remark.ts
@@ -497,7 +497,7 @@ function getFrontmatterNodeValue(file: VFile, obsidianFrontmatter?: ObsidianFron
     frontmatter.tags = obsidianFrontmatter.tags
   }
 
-  return yaml.stringify(frontmatter).trim()
+  return yaml.stringify(frontmatter, { version: '1.1' }).trim()
 }
 
 function getFileUrl(output: StarlightObsidianConfig['output'], filePath: string, anchor?: string) {

--- a/packages/starlight-obsidian/tests/markdown.test.ts
+++ b/packages/starlight-obsidian/tests/markdown.test.ts
@@ -39,6 +39,10 @@ test('sets the proper title', async () => {
   result = await transformFixtureMdFile('basics', 'Basic syntax (comments).md', { includeFrontmatter: true })
 
   expect(result.content).toMatch(/^title: Basic syntax \(comments\)$/m)
+
+  result = await transformFixtureMdFile('basics', '2024-08-19.md', { includeFrontmatter: true })
+
+  expect(result.content).toMatch(/^title: "2024-08-19"$/m)
 })
 
 test('disables edit links', async () => {

